src/index.ts

```diff
@@ -200,27 +200,35 @@ class MagicNixCacheAction extends DetSysAction {
       };
     }
 
-    const notifyPort = inputs.getString("startup-notification-port");
-
-    const notifyPromise = new Promise<Promise<void>>((resolveListening) => {
-      const promise = new Promise<void>(async (resolveQuit) => {
-        const notifyServer = http.createServer((req, res) => {
-          if (req.method === "POST" && req.url === "/") {
-            actionsCore.debug(`Notify server shutting down.`);
-            res.writeHead(200, { "Content-Type": "application/json" });
-            res.end("{}");
-            notifyServer.close(() => {
-              resolveQuit();
-            });
-          }
+    const notifyPromise = new Promise<[Promise<void>, string]>(
+      (resolveListening) => {
+        const promise = new Promise<void>(async (resolveQuit) => {
+          const notifyServer = http.createServer((req, res) => {
+            if (req.method === "POST" && req.url === "/") {
+              actionsCore.debug(`Notify server shutting down.`);
+              res.writeHead(200, { "Content-Type": "application/json" });
+              res.end("{}");
+              notifyServer.close(() => {
+                resolveQuit();
+              });
+            }
+          });
+
+          notifyServer.listen(
+            inputs.getString("startup-notification-port"),
+            () => {
+              actionsCore.debug(`Notify server running.`);
+              const addr = notifyServer.address();
+              if (typeof addr === "string") {
+                resolveListening([promise, addr]);
+              } else if (addr !== null) {
```

Comment:
If `addr` is null, this never resolves and will result in the action hanging. Consider adding a `reject` to the promise constructor so that we can just crash if we don't have something to work with.